.github/workflows/ci.yml

```diff
@@ -33,29 +33,12 @@ jobs:
       matrix:
         os: [ ubuntu-latest, macOS-latest ]
         java: [ 8, 11 ]
-        language: ['java', 'go']
     runs-on: ${{ matrix.os }}
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
 
-      - name: Initialize CodeQL
-        uses: github/codeql-action/init@v2
-        with:
-          # If you wish to specify custom queries, you can do so here or in a config file.
-          # By default, queries listed here will override any specified in a config file.
-          # Prefix the list here with "+" to use these queries and those in the config file.
-          # queries: ./path/to/local/query, your-org/your-repo/queries@main
-          languages: ${{ matrix.language }}
-
-      - if: matrix.language == 'cpp' || matrix.language == 'csharp'
```

Comment:
This step seemed like it would never execute, as neither of these languages was in the matrix, so I did not migrate it to the CodeQL workflow.

Reply:
You mean L52 `if: matrix.language == 'cpp' || matrix.language == 'csharp'` or L50 `languages: ${{ matrix.language }}`?

Reply:
`Build C` task was added in https://github.com/apache/eventmesh/pull/4543. It seems intentional to skip the Build C task, and there is further room for optimization (such as `submodules: true` not added yet) in this area. We have Git submodules in our C SDK, and we want to test these C codes.

Reply:
> You mean L52~L57 task

Yes, the conditional check for cpp/csharp

> Build C task was added in https://github.com/apache/eventmesh/pull/4543. It seems intentional to skip the Build C task, and there is further room for optimization (such as submodules: true not added yet) in this area. We have Git submodules in our C SDK, and we want to test these C codes.

I'm not quite sure what you mean here, since the way this is currently authored, matrix.language will never be either of these two options. So this is essentially a dead code path as it exists today. But I can add it back if it is an area of future change.

Reply:
I have tried to run `Build C` with an additional `submodule: true` param and it encountered a compile error: https://github.com/Pil0tXia/eventmesh/actions/runs/7506142011/job/20436880958. Let's keep the original `Build C` task and I'll address it in another issue.

@xwm1992 Do we still want a `Build C SDK` task?